src/SwedbankPay.Sdk/Common/Amount.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Globalization;
+
+namespace SwedbankPay.Sdk.Common
+{
+    public class Amount : IEquatable<Amount>, IComparable<Amount>, IComparable
+    {
+        private readonly decimal amount;
+
+        public Amount(decimal amount)
+        {
+            this.amount = amount;
+
+            // Use "Banker's Rounding" by default.
+            const MidpointRounding roundingMode = MidpointRounding.ToEven;
+            var roundedAmount = Math.Round(amount, 2, roundingMode);
+            InLowestMonetaryUnit = (int)(roundedAmount * 100);
+        }
+
+        public int InLowestMonetaryUnit { get; }
+
+        public static implicit operator decimal(Amount amount)
+        {
+            return amount.amount;
+        }
+
+        public static implicit operator Amount(decimal amount)
+        {
+            return new Amount(amount);
+        }
+
+        public static Amount operator +(Amount a, Amount b)
+        {
+            return new Amount(a.amount + b.amount);
+        }
+
+        public static Amount operator -(Amount a, Amount b)
+        {
+            return new Amount(a.amount - b.amount);
+        }
+
+        public static Amount FromLowestMonetaryUnit(int amountInLowestMonetaryUnit)
+        {
+            var amount = (decimal)amountInLowestMonetaryUnit / 100;
+            return new Amount(amount);
+        }
+
+        public int CompareTo(object obj)
+        {
+            if (obj is null)
+                return 1;
+
+            if (ReferenceEquals(this, obj))
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)